src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_FindUsages.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    // root level service for all Roslyn services
+    internal partial class CodeAnalysisService : IRemoteFindUsagesService
+    {
+        public Task FindReferencesAsync(PinnedSolutionInfo solutionInfo,
+            SerializableSymbolAndProjectId symbolAndProjectIdArg,
+            SerializableFindReferencesSearchOptions options,
+            CancellationToken cancellationToken)
+        {
+            return RunServiceAsync(async () =>
+            {
+                using (UserOperationBooster.Boost())
+                {
+                    var solution = await GetSolutionAsync(solutionInfo, cancellationToken).ConfigureAwait(false);
+
+                    var symbol = await symbolAndProjectIdArg.TryRehydrateAsync(
+                        solution, cancellationToken).ConfigureAwait(false);
+                    var project = solution.GetProject(symbolAndProjectIdArg.ProjectId);
+
+                    var context = new RemoteFindUsageContext(solution, EndPoint, cancellationToken);
+
+                    if (symbol == null)
+                        return;
+
+                    await AbstractFindUsagesService.FindReferencesAsync(
+                        context, symbol, project, options.Rehydrate()).ConfigureAwait(false);
+                }
+            }, cancellationToken);
+        }
+
+
+        public Task FindImplementationsAsync(PinnedSolutionInfo solutionInfo,
```

Comment:
>PinnedSolutionInf [](start = 45, length = 17)

Nit: this is an odd indenting, i'd expect the first param to be on the next line.